src/encord_active/lib/model_predictions/importers.py

```diff
@@ -55,214 +57,504 @@
     ("rotation_y", float),  # [-pi, pi]
 ]
 
-KITTI_FILE_NAME_REGEX = r"^(?P<data_hash>[a-f0-9\-]*)__(?P<image_name>.*)$"
-PNG_FILE_NAME_REGEX = r"^(?P<stem>.*?)\.png$"
+KITTI_FILE_NAME_REGEX = r"^(?P<data_unit_title>.*?)(?:__(?P<frame>\d+))?\.(?:txt|csv)$"
+PNG_FILE_NAME_REGEX = r"^(?P<data_unit_title>.*?)(?:__(?P<frame>\d+))?\.png$"
 
 
-def import_mask_predictions(
-    project: EncordProject,
-    data_root: Path,
-    cache_dir: Path,
-    prediction_writer: PredictionWriter,
-    class_map: Optional[Dict[str, int]] = None,
-    file_name_regex: str = PNG_FILE_NAME_REGEX,
-    du_hash_name_lookup: Callable[[Path], Tuple[str, int]] = None,
-):
+def import_predictions(project: Project, predictions: list[Prediction]):
     """
-    Will look for mask
-    :param project: The project for which to which the predictions are associated
-    :param data_root: The root folder of the segmentation masks
-    :param cache_dir: The cache directory of the (potentially) cached label rows, images, etc.
-    :param prediction_writer: The writer to store the predictions with.
-    :param class_map: A Dictionary which contains ``(featureNodeHash, class_idx)`` pairs, such that if
-        a pixel in a mask has value 1 and corresponds to ``featureNodeHash == "abcdef01``, then
-        ``class_map["abcdef01"] == 1``.
-        If the ``class_map`` is not specified, it will be generated by::
-
-            polygon_classes = sorted({
-                    o for o in OntologyStructure.from_dict(project.ontology).objects if o.shape == Shape.POLYGON
-                },
-                key=lambda x: int(x.id)
-            )
-            {o.feature_node_hash: i+1 for o in enumerate(polygon_classes) if o.shape == "polygon"}
-
-    :param file_name_regex: A regex pattern to match annotation masks. As default, all pngs in the `data_root`
-        will be considered as predictions. Example strings that are matched are::
+    Import predictions from Encord's Prediction class format into the specified Encord Active project.
 
-            "/path/to/data_root/masks/some_name.png"
-            "/path/to/data_root/some_name.png"
+    :param project: The project to import the predictions into.
+    :param predictions: The predictions in Encord's Prediction class format to be imported.
+    """
+    if len(predictions) == 0:
+        return
 
-    :param du_hash_name_lookup: A transform ``f(file_path) -> (data_unit_hash: str, frame: int)`` to translate between
-        data units and their associated mask file names. You can omit this argument if you
+    with PredictionWriter(project) as writer:
```

Comment:
The importer will break if there are different types of predictions in the same import. I would add a check up front here and check two things.

1. Assert that all predictions are either classifications or objects.
2. If they are all objects, ensure that the `prediction.format` is consistent across all predictions

If that's not the case throw a meaningful error such that users know what the limitations / issues are

Reply:
On point 2, seems like the original COCO predictions importer sent predictions with two object formats, `polygon` and `bbox`, so probably we should continue allowing the same behaviour.